fixed: Web - Desktop Attachments - Cannot attach a photo when the NEW ( green plus) button is pressed

metehanozyurt opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
<!-- Explanation of the change or anything fishy that is going on -->
When user click Create icon open menu and events drag/drop wont work. This PR fix this issue.

### Fixed Issues
<!---
Please replace GH_LINK with the link to the GitHub issue this Pull Request is fixing.
Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the issue; otherwise, the linking will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<number-of-the-issue>

Do NOT only link the issue number like this: $ #<number-of-the-issue>
--->
$ https://github.com/Expensify/App/issues/8842

### Tests
<!---
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->
1. Open Expensify account on Web
2. Select any chat room in your list
3. Tap the NEW (green plus) button on the bottom left of the screen
4. Open the folder with the desired file on your computer
5. Drag a photo from the folder to the chat room
6. Notice that opened menu closed and user can drop file to the chat
6. Open Expensify account on Web
7. Select any chat room in your list
8. Tap the NEW (green plus) button on the bottom left of the screen
9. Open the folder with the desired file on your computer
10. Drag a photo from the folder to the chat room
11. Notice that opened menu closed and user can drop file to the chat
12. mWebs, ios native, android native are not affected by this PR.


- [x] Verify that no errors appear in the JS console

### PR Review Checklist
<!--
This is a checklist for PR authors & reviewers. Please make sure to complete all tasks and check them off once you do, or else Expensify has the right not to merge your PR!
-->
#### Contributor (PR Author) Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [ ] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [ ] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose and it is
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn’t already exist
    - [ ] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.


<details>
<summary><h4>PR Reviewer Checklist</h4></summary>

- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn’t already exist
    - [ ] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

</details>

### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

1. Open Expensify account on Web
2. Select any chat room in your list
3. Tap the NEW (green plus) button on the bottom left of the screen
4. Open the folder with the desired file on your computer
5. Drag a photo from the folder to the chat room
6. Notice that opened menu closed and user can drop file to the chat
6. Open Expensify account on Web
7. Select any chat room in your list
8. Tap the NEW (green plus) button on the bottom left of the screen
9. Open the folder with the desired file on your computer
10. Drag a photo from the folder to the chat room
11. Notice that opened menu closed and user can drop file to the chat
12. mWebs, ios native, android native are not affected by this PR.

- [ ] Verify that no errors appear in the JS console

### Screenshots
<!-- Add screenshots for all platforms tested. Pull requests won't be merged unless the screenshots show the app was tested on all platforms.-->

#### Web
<!-- Insert screenshots of your changes on the web platform-->


https://user-images.githubusercontent.com/6093207/168465272-10c8d200-492f-4fcb-9ec3-d32a9d7fa899.mp4


#### Mobile Web
<!-- Insert screenshots of your changes on the web platform (from a mobile browser)-->


https://user-images.githubusercontent.com/6093207/168465330-52bc94d6-b742-45b8-9923-9f32e7233c83.mp4


https://user-images.githubusercontent.com/6093207/168465202-4d93d0e9-1d41-494e-950e-a0a71be2b5ee.mp4


#### Desktop
<!-- Insert screenshots of your changes on the desktop platform-->


https://user-images.githubusercontent.com/6093207/168465264-c393b5f0-27a7-4d6f-9fb8-5ec76328f40e.mp4


#### iOS
<!-- Insert screenshots of your changes on the iOS platform-->

https://user-images.githubusercontent.com/6093207/168465322-141b7b11-83dd-4f45-801f-b2d4790005d0.mp4


#### Android
<!-- Insert screenshots of your changes on the Android platform-->

https://user-images.githubusercontent.com/6093207/168465190-e2a3c723-13f0-4eb7-924d-30ad21682e54.mp4

metehanozyurt committed:
dragOverListener method added to SidebarScreen

metehanozyurt committed:
document changed to window.document and conditions added for unnecessary bind and listeners

mananjadhav commented on src/pages/home/sidebar/SidebarScreen.js:
```diff
@@ -83,6 +89,9 @@ class SidebarScreen extends Component {
      */
     hideCreateMenu() {
+        if (window.document) {
+            window.document.removeEventListener('dragover', this.dragOverListener);
```
We need to refactor this component to have web platform-specific changes in the `index.web.js`.

Please check the the guidelines here: https://github.com/Expensify/App#platform-specific-file-extensions

metehanozyurt replied:
Thank you for your comment @mananjadhav . 
When I saw that it didn't work on android and ios, I wanted to do this too. Frankly, I was afraid to repeat too much code for 10 lines. If code duplication is not a problem, I would like to do it as follows, if you see fit. 
I'm a little afraid that the same codes will repeat. I wanted to get your permission before I did it.

![WhatsApp Image 2022-05-15 at 22 22 59](https://user-images.githubusercontent.com/6093207/168490587-67ae6f0c-4e30-4e19-b9ad-b51229a6406c.jpeg)

mananjadhav replied:
In these cases we should push all the common code to a component say `BaseSidebarScreen`. We then add/override only platform-specific functions and noops in others.

A quick small example, would look like this:

```
BaseSidebarScreen.js

const propTypes = {
      beforeHideCreateMenu: PropTypes.func,
}

class BaseSidebarScreen {

    hideCreateMenu() {
       if(this.props.beforeHideCreateMenu) { 
           this.props.beforeHideCreateMenu();
      }. 

       this.setState({
            isCreateMenuActive: false,
        });
    }
   

}

index.web.js
class SidebarScreen {   
    render() {
       return <BaseSidebarScreen ... />
    }
}


`index.js`
class SidebarScreen {   
    render() {
       return <BaseSidebarScreen ... beforeHideCreateMenu={() => { document.removeEventListener('dragover', this.dragOverListener); }} />
    }
}

```

I hope this helps, do reach out if you have any other questions

metehanozyurt replied:
Thank you so much helps a lot 🙏 . I will try to do as you say. thank you so much.

metehanozyurt replied:
I made a change as you said in the light of the guidance. I hope you will like it @mananjadhav 🙏 👼 . 
```js
if (this.props.beforeHideCreateMenu) {
            this.props.beforeHideCreateMenu();
        }
```
```js
if (this.props.afterShowCreateMenu) {
            this.props.afterShowCreateMenu(this);
        }
```
These conditions may have been extra. I need your opinion as I am not sure. Because I defined it as default props.
Thank you for your help again. I wouldn't dare to change the files 😅.

metehanozyurt committed:
BaseSidebarScreen.js, index.js and index.native.js created.

mananjadhav commented on src/pages/home/sidebar/SidebarScreen/BaseSidebarScreen.js:
```diff
@@ -68,6 +77,9 @@ class SidebarScreen extends Component {
         this.setState({
             isCreateMenuActive: true,
         });
+        if (this.props.afterShowCreateMenu) {
```
You can remove this check.

mananjadhav commented on src/pages/home/sidebar/SidebarScreen/BaseSidebarScreen.js:
```diff
@@ -68,6 +77,9 @@ class SidebarScreen extends Component {
             isCreateMenuActive: true,
         });
+        if (this.props.afterShowCreateMenu) {
+            this.props.afterShowCreateMenu(this);
```
I would recommend removing `this` from the parameters. Instead pass a `ref` in the `SidebarScreen/index.js`

mananjadhav commented on src/pages/home/sidebar/SidebarScreen/BaseSidebarScreen.js:
```diff
@@ -96,6 +111,15 @@ class SidebarScreen extends Component {
+     *
+     * @param {Object} e native Event
+     */
+    dragOverListener() {
```
Is this function required here?

mananjadhav commented on src/pages/home/sidebar/SidebarScreen/index.js:
```diff
@@ -0,0 +1,40 @@
+     * @param {BaseSidebarScreen} baseComponent
+     */
+    createDragoverListener= (baseComponent) => {
+        this.dragOverListener = this.dragOverListener.bind(baseComponent);
```
Don't pass `baseComponent`. Instead add a ref to `BaseSidebarScreen` and use that.

mananjadhav commented on src/pages/home/sidebar/SidebarScreen/index.js:
```diff
@@ -0,0 +1,40 @@
+import React, {PureComponent} from 'react';
+import BaseSidebarScreen from './BaseSidebarScreen';
+
+class SidebarScreen extends PureComponent {
```
I think we can change this `Function` component. We're not using `setState` and hence don't need this to be a `Class Component`

mananjadhav commented on src/pages/home/sidebar/SidebarScreen/BaseSidebarScreen.js:
```diff
@@ -5,26 +5,33 @@ import {View} from 'react-native';
 const propTypes = {
+
+    /* Create Listener callback */
+    afterShowCreateMenu: PropTypes.func,
```
I know I suggested this name, but generally, I avoid `before` or `after` naming conventions, and rather use `onHide`, etc. Let me think if we can name it better. Do let me know if you have better suggestions.

mananjadhav reviewed (changes requested):
Thanks for the changes @metehanozyurt. Looks better, have left a few comments. 

Do ping if you need any help.

metehanozyurt committed:
index.js and index.native.js files changed to function component, index.js using ref now

metehanozyurt commented:
I'm so sorry for being late @mananjadhav.
* Removed unnecessary if conditions.
* Class components changed to function components.
* Used for ref to call `hideCreateMenu` function. 
* Deleted `dragOverListener` function on `BaseSidebarScreen` and `index` and  connect ref instead.
* Renamed functions to `onHideCreateMenu` and `onShowCreateMenu` now.

I hope you like these changes. Thank you very much for your time and helps 🙏 👼 .

mananjadhav commented on src/pages/home/sidebar/SidebarScreen/index.native.js:
```diff
@@ -0,0 +1,7 @@
+import React from 'react';
+import BaseSidebarScreen from './BaseSidebarScreen';
+
+export default function SidebarScreen(props) {
```
Add `displayName`, `propTypes` for `SidebarScreen`.

metehanozyurt replied:
When I add propTypes I had to add this lines so that the required fields come . But I removed it from baseSidebar.
```js
export default compose(
    withNavigation,
    withLocalize,
    withWindowDimensions,
    withOnyx({
        allPolicies: {
            key: ONYXKEYS.COLLECTION.POLICY,
        },
        betas: {
            key: ONYXKEYS.BETAS,
        },
        isCreatingWorkspace: {
            key: ONYXKEYS.IS_CREATING_WORKSPACE,
        },
    }),
)(SidebarScreen);
```

mananjadhav commented on src/pages/home/sidebar/SidebarScreen/index.js:
```diff
@@ -0,0 +1,29 @@
+import React, {useRef} from 'react';
+import BaseSidebarScreen from './BaseSidebarScreen';
+
+export default function (props) {
```
Name the function

metehanozyurt replied:
I am very embarrassed for the mistake I made. Please accept my apology :( .

mananjadhav replied:
No worries. Happens with the best of us.

metehanozyurt committed:
sidebarPropTypes.js created,displayName and propTypes added for SidebarScreen.

mananjadhav commented on src/pages/home/sidebar/SidebarScreen/BaseSidebarScreen.js:
```diff
@@ -2,29 +2,32 @@ import lodashGet from 'lodash/get';
+    onHideCreateMenu: PropTypes.func,
+
     /* Beta features list */
     betas: PropTypes.arrayOf(PropTypes.string).isRequired,
```
This can come from the `sidebarPropTypes`

mananjadhav commented on src/pages/home/sidebar/SidebarScreen/BaseSidebarScreen.js:
```diff
@@ -59,6 +64,9 @@ class SidebarScreen extends Component {
 
         const routes = lodashGet(this.props.navigation.getState(), 'routes', []);
         Welcome.show({routes, showCreateMenu: this.showCreateMenu});
+        if (this.props.innerRef) {
```
We don't need the `ref` here. Please check my comment below.

mananjadhav commented on src/pages/home/sidebar/SidebarScreen/index.js:
```diff
@@ -0,0 +1,56 @@
+import BaseSidebarScreen from './BaseSidebarScreen';
+
+const SidebarScreen = (props) => {
+    const BaseSidebarScreenRef = useRef(null);
```
```suggestion
    let baseSidebarScreen = null;
```

mananjadhav commented on src/pages/home/sidebar/SidebarScreen/index.js:
```diff
@@ -0,0 +1,56 @@
+     * Method create event listener
+     */
+    const createDragoverListener = () => {
+        document.addEventListener('dragover', BaseSidebarScreenRef.current.hideCreateMenu);
```
```suggestion
        document.addEventListener('dragover', baseSidebarScreen.hideCreateMenu);
```

mananjadhav commented on src/pages/home/sidebar/SidebarScreen/index.js:
```diff
@@ -0,0 +1,56 @@
+     * Method remove event listener.
+     */
+    const removeDragoverListener = () => {
+        document.removeEventListener('dragover', BaseSidebarScreenRef.current.hideCreateMenu);
```
```suggestion
        document.removeEventListener('dragover', baseSidebarScreen.hideCreateMenu);
```

mananjadhav commented on src/pages/home/sidebar/SidebarScreen/index.js:
```diff
@@ -0,0 +1,56 @@
+    };
+    return (
+        <BaseSidebarScreen
+            innerRef={BaseSidebarScreenRef}
```
```suggestion
            ref={el => baseSidebarScreen = el}
```

mananjadhav commented on src/pages/home/sidebar/SidebarScreen/index.js:
```diff
@@ -0,0 +1,56 @@
+};
+
+SidebarScreen.propTypes = propTypes;
+SidebarScreen.displayName = 'SidebarScreen';
```
Might want to add `defaultProps`

mananjadhav commented on src/pages/home/sidebar/SidebarScreen/index.native.js:
```diff
@@ -0,0 +1,32 @@
+const SidebarScreen = props => <BaseSidebarScreen {...props} />;
+
+SidebarScreen.propTypes = propTypes;
+SidebarScreen.displayName = 'SidebarScreen';
```
Might want to add `defaultProps`

mananjadhav commented on src/pages/home/sidebar/SidebarScreen/sidebarPropTypes.js:
```diff
@@ -0,0 +1,17 @@
+
+const propTypes = {
+    /* Beta features list */
+    betas: PropTypes.arrayOf(PropTypes.string).isRequired,
```
I know this was missing earlier as well, but if you see we're adding `allPolicies` as well in the props via Onyx. So we would need to add it in the `propTypes` too.

mananjadhav reviewed (changes requested):
PR has made good progress. Thanks, @metehanozyurt. Few more suggestions and we should be good to merge.

metehanozyurt committed:
allPolicies added to sidebarPropTypes. Fixed the use of ref. BaseSidebarScreen using sidebarPropTypes.

metehanozyurt commented:
I made the changes based on your suggestions @mananjadhav . Thank you very much for your valuable time and support 🙏 . You can be sure that I will do my best not to make the mistakes I have made in my next works.

```js
/* Create Listener callback */
    onShowCreateMenu: PropTypes.func,

    /* Remove Listener callback */
    onHideCreateMenu: PropTypes.func,
```
I did not move these two properties into sidebarPropTypes. I did it like this because it is not used in native code. If you want me to do that, I can do it right away.

Luke9389 commented on src/pages/home/sidebar/SidebarScreen/BaseSidebarScreen.js:
```diff
@@ -2,44 +2,40 @@ import lodashGet from 'lodash/get';
 
-    /* Is workspace is being created by the user? */
-    isCreatingWorkspace: PropTypes.bool,
+    /* Create Listener callback */
```
This could be more descriptive

Luke9389 commented on src/pages/home/sidebar/SidebarScreen/index.js:
```diff
@@ -0,0 +1,57 @@
+SidebarScreen.defaultProps = sidebarDefaultProps;
+SidebarScreen.displayName = 'SidebarScreen';
+
+export default compose(
+    withNavigation,
+    withLocalize,
+    withWindowDimensions,
+    withOnyx({
+        allPolicies: {
+            key: ONYXKEYS.COLLECTION.POLICY,
+        },
+        betas: {
+            key: ONYXKEYS.BETAS,
+        },
+        isCreatingWorkspace: {
+            key: ONYXKEYS.IS_CREATING_WORKSPACE,
+        },
+    }),
+)(SidebarScreen);
```
Can we compose in the BaseSidebarScreen instead of doing it in each of the SidebarScreen components?

mananjadhav replied:
Because we use the `ref` it gives an error with `withOnyx` forward reference,

Luke9389 replied:
Oh, I see. Thanks @mananjadhav 🙇

metehanozyurt committed:
descriptive comment added.

mananjadhav commented on src/pages/home/sidebar/SidebarScreen/BaseSidebarScreen.js:
```diff
@@ -2,44 +2,40 @@ import lodashGet from 'lodash/get';
 
-    /* Is workspace is being created by the user? */
-    isCreatingWorkspace: PropTypes.bool,
+    /* A callback to call when we show create menu */
```
Callback function when the menu is shown

mananjadhav commented on src/pages/home/sidebar/SidebarScreen/BaseSidebarScreen.js:
```diff
@@ -2,44 +2,40 @@ import lodashGet from 'lodash/get';
 
-    ...windowDimensionsPropTypes,
+    /* A callback to call when we hide create menu */
+    onHideCreateMenu: PropTypes.func,
```
Callback function before the menu is hidden

mananjadhav reviewed (changes requested):
@metehanozyurt Sorry for the delay. I've slightly changed the language of the comment in the suggestion. Rest all it looks good to me.

Also can you pull the latest main branch?

mananjadhav commented on src/pages/home/sidebar/SidebarScreen/BaseSidebarScreen.js:
```diff
@@ -2,44 +2,40 @@ import lodashGet from 'lodash/get';
+    onShowCreateMenu: PropTypes.func,
 
-    ...windowDimensionsPropTypes,
+    /* A callback to call when we hide create menu */
```
```suggestion
    /* Callback function before the menu is hidden*/
```

metehanozyurt committed:
comments changed

metehanozyurt committed:
Merge branch 'main' of https://github.com/Expensify/App into issue-8842

metehanozyurt commented:
Thank you @mananjadhav I changed comments and pulled the latest main branch.
but Some checks were not successful . Is there anything I can do about this? It didn't give an error in the code I added 😟.

mananjadhav commented:
Changes look fine, but I am not sure about the break in the GH action. Neither do I have an option to rerun the workflow. @Luke9389 can you help here?

PR Review done

🎀 👀 🎀
C+ reviewed

Luke9389 commented:
Re-ran the GH action and it passed ✅
I'll review this first thing tomorrow. Thanks y'all!

Luke9389 reviewed (approved):
lookin good, thanks!

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/2462844753) to staging by @Luke9389 in version: 1.1.74-0 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|failure ❌ 
🖥 desktop 🖥|failure ❌
🍎 iOS 🍎|cancelled 🔪 
🕸 web 🕸|failure ❌
